src/plugins/trace/exe/Program.cs

```diff
@@ -232,6 +238,35 @@ public void UpdateToState(RequestState state, ulong time, bool ignorePrevious =
                 State = state;
             }
 
+            public void UpdateToIdle(ulong time)
```

Comment:
Why a separate UpdateToIdle method when there's an UpdateToState method and "Idle" is just another RequestState value?

Reply:
To be clear this is updating to **an** idle state, but not a specific one, because it's dependent on the current state of send/recv.

Reply:
`Idle` is equivalent to "idle with no send or recv yet" but I didn't want to write that all in an enum. This code isn't just changing to that state, but one of 4:
```c
                if (FirstPacketRecv != 0 && FirstPacketSend != 0)
                {
                    State = RequestState.IdleBoth;
                }
                else if (FirstPacketRecv != 0)
                {
                    State = RequestState.IdleRecv;
                }
                else if (FirstPacketSend != 0)
                {
                    State = RequestState.IdleSent;
                }
                else
                {
                    State = RequestState.Idle;
                }
```

Reply:
Gotcha.